qiskit/compiler/transpiler.py

```diff
@@ -356,76 +363,52 @@ def callback_func(**kwargs):
-                    inst_map=inst_map,
-                    coupling_map=coupling_map,
-                    instruction_durations=instruction_durations,
-                    backend_properties=backend_properties,
-                    timing_constraints=timing_constraints,
-                    initial_layout=initial_layout,
-                    layout_method=layout_method,
-                    routing_method=routing_method,
-                    translation_method=translation_method,
-                    scheduling_method=scheduling_method,
-                    approximation_degree=approximation_degree,
-                    seed_transpiler=seed_transpiler,
-                    unitary_synthesis_method=unitary_synthesis_method,
-                    unitary_synthesis_plugin_config=unitary_synthesis_plugin_config,
-                    hls_config=hls_config,
-                    init_method=init_method,
-                    optimization_method=optimization_method,
-                    _skip_target=_skip_target,
-                )
-                out_circuits.append(pm.run(circuit, callback=callback))
-            for name, circ in zip(output_name, out_circuits):
-                circ.name = name
-                end_time = time()
-            _log_transpile_time(start_time, end_time)
-            return out_circuits
-        else:
-            instruction_durations = _parse_instruction_durations(
-                backend, instruction_durations, dt, circuits[0]
+        for i in range(0, len(circuits)):
+            instruction_duration = _parse_instruction_durations(
+                backend, instruction_durations, dt, circuits[i]
             )
-
-    pm = generate_preset_pass_manager(
-        optimization_level,
-        backend=backend,
-        target=target,
-        basis_gates=basis_gates,
-        inst_map=inst_map,
-        coupling_map=coupling_map,
-        instruction_durations=instruction_durations,
-        backend_properties=backend_properties,
-        timing_constraints=timing_constraints,
-        initial_layout=initial_layout,
-        layout_method=layout_method,
-        routing_method=routing_method,
-        translation_method=translation_method,
-        scheduling_method=scheduling_method,
-        approximation_degree=approximation_degree,
-        seed_transpiler=seed_transpiler,
-        unitary_synthesis_method=unitary_synthesis_method,
-        unitary_synthesis_plugin_config=unitary_synthesis_plugin_config,
-        hls_config=hls_config,
-        init_method=init_method,
-        optimization_method=optimization_method,
-        _skip_target=_skip_target,
+            pair = (circuits[i], instruction_duration)
+            circuits_instruction_durations.append(pair)
+    else:
+        for i in range(0, len(circuits)):
+            pair = (circuits[i], None)
+            circuits_instruction_durations.append(pair)
+
+    out_circuits = []
+    out_circuits = parallel_map(
+        _run_circuits,
```

Comment:
This is potentially problematic as we're moving the unit of parallelism from being `PassManager.run` to the outer function that builds the pass manager and then runs it in a parallel context. This is actually how things used to be setup in `transpile` before and we moved to building a single passmanager and doing the parallel dispatch inside `.run()` both for consistency in the execution paths and also ebcause it's lower overhead for most multiprocessing modes. Using `PassManager.run()` avoids a bunch of serialization and IPC as `task_args` have to be directly serialized and sent via IPC to the child processes.

Reply:
Ok that makes sense I will update accordingly.

Reply:
Changes have been made to pass the num_processes through PassManager.run